packages/babel-helper-remap-async-to-generator/src/index.js

```diff
@@ -83,12 +83,15 @@ function classOrObjectMethod(path: NodePath, callId: Object) {
 
   node.async = false;
 
-  const container = t.functionExpression(null, [], t.blockStatement(body.body), true);
+  const container = t.functionExpression(node.key, [], t.blockStatement(body.body), true);
```

Comment:
Hmm, I hadn't considered those cases. It simplest way to handle those are to test if the method name is a valid identifier and not use it if it isn't. Alternatively, we could pass in non-identifiers (computed properties) as a second param to `asyncToGenerator`, which would use `Object.defineProperty` to set the function name. That would have a performance penalty for non-valid function names, but would be the most complete.